src/features/messages/cards/GasDetailsCard.tsx

```diff
@@ -29,14 +29,20 @@ export function GasDetailsCard({ message, blur, igpPayments = {} }: Props) {
   const unitOptions = useMemo(() => {
     const originMetadata = multiProvider.tryGetChainMetadata(message.originDomainId);
     const nativeCurrencyName = originMetadata?.nativeToken?.symbol || 'Eth';
+    const nativeDecimals = originMetadata?.nativeToken?.decimals || 18;
     return [
-      { value: 18, display: toTitleCase(nativeCurrencyName) },
+      { value: nativeDecimals, display: toTitleCase(nativeCurrencyName) },
       { value: 9, display: 'Gwei' },
       { value: 0, display: 'Wei' },
     ];
   }, [message, multiProvider]);
 
-  const [decimals, setDecimals] = useState(unitOptions[0].value);
+  const [decimals, setDecimals] = useState<number>(unitOptions[0].value);
+
+  // Update `decimals` after the component mounts to ensure it matches the initial `unitOptions` value
+  useEffect(() => {
+    setDecimals(unitOptions[0].value);
+  }, [unitOptions]);
```

Comment:
Is this hook necessary? `unitOptions[0].value` will already be the default value for that state because it's set here: `useState<number>(unitOptions[0].value);`

This will only have an effect if the `unitOptions` value changes, like if the multiProvider changes. If that's the case, please update the comment